### PR TITLE
QueryOperationRow: Removed overflow from title and added scroll to DetailText

### LINKED ISSUE #32893: QueryOperationRow: headerElement overflows the row and hides title
**What happened**:
When **headerElement** is passed to **QueryOperationRow**, if it is longer than available space, it:
1. Hides the title of QueryOperationRow(see that Data option is not visible):
image

2. Overflows the row (see the text behind Download button): 
image

**What you expected to happen**:
I would expect that **title** is always visible as it seems more important. For the headerElement, I would expect for overflow to be hidden or ended with ellipses or it would be wrapped or had horizontal scrollbar. 

**How to reproduce it (as minimally and precisely as possible)**:
Run prometheus query (or other query that returns multiple dataFrames)  `alertmanager_alerts` -> open the Panel inspect -> Data tab.

### PR DESCRIPTION
**Pull request by grpit**
**What this PR does / why we need it**:
On Inspect panel the title was not shown when space wasn't available. Removed overflow hidden to fix this.
The Query was overflown, overflow set to scroll and scrollbar is hidden as it was blocking the text.

**Which issue(s) this PR fixes**:

Fixes #32893

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by torkelo** [comment]
can you show what you mean?  if there is not enough space for the text is should be hidden/ellipsis

**Comment by grpit** [comment]
https://user-images.githubusercontent.com/76885189/115136730-61418600-a03f-11eb-8145-4a50e008e98c.mov

Looks something like this. If there is no space the text is hidden but user can scroll over it to see the full text.

**Comment by torkelo** [comment]
Not sure making this area scrollable makes sense

**Comment by grpit** [comment]
I thought it would be better experience if the user can see whole text with scroll.
Removing it in favour of ellipsis.

Screenshot 2021-04-18 at 4 24 52 PM

**Comment by mckn** [comment]
> I thought it would be better experience if the user can see whole text with scroll.
> Removing it in favour of ellipsis.
> 
> Screenshot 2021-04-18 at 4 24 52 PM

Thanks for opening this PR! However to make the UI consistent we should probably end the text with `text that is loo...` if it is too long but also add an `alt` or a `title` attribute so you can put the cursor on top of the text to view the full text when needed.

**Comment by grpit** [comment]
How should I go about adding the title ?
The paragraph is rendered from [this](https://github.com/grafana/grafana/blob/4b150297c0eb3484fa40c977f81b818d04fb1e49/public/app/features/inspector/DetailText.tsx#L15) component. Should I add a title prop to the component or adding `title={typeof children === 'string' ? children : ''}` in the `p` tag seems okay ?

**Comment by mckn** [comment]
> How should I go about adding the title ?
> The paragraph is rendered from [this](https://github.com/grafana/grafana/blob/4b150297c0eb3484fa40c977f81b818d04fb1e49/public/app/features/inspector/DetailText.tsx#L15) component. Should I add a title prop to the component or adding `title={typeof children === 'string' ? children : ''}` in the `p` tag seems okay ?

Yeah, that should work 👍🏻 

@torkelo what do you think about the approach with the `title` attribute?

**Comment by dprokop** [comment]
> How should I go about adding the title ?
> The paragraph is rendered from this component. Should I add a title prop to the component or adding title={typeof children === 'string' ? children : ''} in the p tag seems okay ?

I suggest changing the interface of [DetailText](https://github.com/grafana/grafana/blob/4b150297c0eb3484fa40c977f81b818d04fb1e49/public/app/features/inspector/DetailText.tsx#L15) to enforce children to be a string.

```
interface DetailTextProps {
  children: string;
}

export function DetailText({ children }: DetailTextProps) {
 ...
```

I don't see we use DetailText in any other context than providing a string.

**Comment by grpit** [comment]
Updated code based on comments from @dprokop 
cc @mckn @torkelo

File: public/app/core/components/QueryOperationRow/QueryOperationRow.tsx [hunk 1]
```diff
@@ -154,7 +154,6 @@ const getQueryOperationRowStyles = stylesFactory((theme: GrafanaTheme) => {
       align-items: center;
       flex-grow: 1;
       cursor: pointer;
-      overflow: hidden;
```
Comment (torkelo) [inline public/app/core/components/QueryOperationRow/QueryOperationRow.tsx:157]: this breaks other uses of this component

Comment (torkelo) [reply public/app/core/components/QueryOperationRow/QueryOperationRow.tsx:157]: like QueryEditors and TransformationRows & QueryOptions

Comment (grpit) [reply public/app/core/components/QueryOperationRow/QueryOperationRow.tsx:157]: Thanks for pointing out. The only other way I could think of was to set `flex-shrink: 100 /* or some other large value */` on [DetailText](https://github.com/grafana/grafana/blob/4b150297c0eb3484fa40c977f81b818d04fb1e49/public/app/features/inspector/DetailText.tsx#L15). Let me know if thats okay or you have any other suggestions.

Comment (mckn) [reply public/app/core/components/QueryOperationRow/QueryOperationRow.tsx:157]: hmm, do you really need to remove the "overflow:hidden" in the solution without the scrolling?

Comment (grpit) [reply public/app/core/components/QueryOperationRow/QueryOperationRow.tsx:157]: With "overflow:hidden" the title wrapper shrinks and hides the text.

image


Removing overflow breaks other uses of the component. flex-shrink on the DetailText element was the only alternative I could come up with, which requires minimal changes.

**Pull request closed without merging**